### PR TITLE
Update skipped azkeys tests

### PR DESCRIPTION
**Pull request by chlowell**
Fixing some test setup issues and unskipping and re-recording rotation policy and secure key release tests.

### CODE REVIEW

File: sdk/keyvault/azkeys/utils_test.go [hunk 1]
```diff
@@ -22,15 +23,35 @@ import (
...
+)
 
 func TestMain(m *testing.M) {
-	// Initialize
+	liveVaultURL = strings.TrimSuffix(os.Getenv("AZURE_KEYVAULT_URL"), "/")
+	liveMHSMURL = strings.TrimSuffix(os.Getenv("AZURE_MANAGEDHSM_URL"), "/")
+	enableHSM = liveMHSMURL != ""
+
+	if recording.GetRecordMode() != recording.LiveMode {
+		err := recording.ResetProxy(nil)
```
Comment (jhendrixMSFT) [inline sdk/keyvault/azkeys/utils_test.go:44]: Why do we need to do this before/after the tests?  Is this just one of those "ensure the proxy is in a known state" type of thing?

File: sdk/keyvault/azkeys/utils_test.go [hunk 2]
```diff
@@ -40,48 +61,37 @@ func TestMain(m *testing.M) {
...
-
-		err = recording.AddBodyKeySanitizer("$.recoveryId", fakeKvURL, vaultUrl, nil)
-		if err != nil {
-			panic(err)
+		if liveVaultURL == "" {
+			panic("no value for AZURE_KEYVAULT_URL")
 		}
-
-		tenantID := os.Getenv("AZKEYS_TENANT_ID")
-		err = recording.AddHeaderRegexSanitizer("WWW-Authenticate", "00000000-0000-0000-0000-000000000000", tenantID, nil)
```
Comment (jhendrixMSFT) [inline sdk/keyvault/azkeys/utils_test.go:60]: Why was this removed?

File: sdk/keyvault/azkeys/utils_test.go (hunk 1 above)
Comment (chlowell) [reply sdk/keyvault/azkeys/utils_test.go:44]: The latter. Just trying to prevent hard to debug problems when working on multiple modules.

File: sdk/keyvault/azkeys/utils_test.go (hunk 2 above)
Comment (chlowell) [reply sdk/keyvault/azkeys/utils_test.go:60]: There's no need to replace tenant IDs because they aren't secret or considered in request matching. However, when the variable isn't set and `tenantID == ""`, this sanitizer will malform the header and break playback.

Comment (jhendrixMSFT) [reply sdk/keyvault/azkeys/utils_test.go:60]: For the first part, that makes sense.

For the second part, sounds like this is something we should fix in `recording`.

**Review by heaths — APPROVED** [review]

**Review by jhendrixMSFT — APPROVED** [review]
Get a tracking issue opened for the problem with `recording`.

File: sdk/keyvault/azkeys/utils_test.go (hunk 2 above)
Comment (chlowell) [reply sdk/keyvault/azkeys/utils_test.go:60]: #17842
